archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/internal/ArchUnitTestDescriptor.java

```diff
@@ -295,11 +300,14 @@ private static class JUnit5ClassAnalysisRequest implements ClassAnalysisRequest
         }
 
         private static AnalyzeClasses checkAnnotation(Class<?> testClass) {
-            AnalyzeClasses analyzeClasses = testClass.getAnnotation(AnalyzeClasses.class);
-            checkArgument(analyzeClasses != null,
+            List<AnalyzeClasses> analyzeClasses = new AnnotationFinder<>(AnalyzeClasses.class).findAnnotationsOn(testClass);
```

Comment:
Not sure how to reach this from test.

Reply:
This only gets called when you execute the discovery request. So, all the execute unit tests of `ArchUnitTestEngine` cover this, you can just destroy the place and run the test 😉

Reply:
Not quite sure if I understand it correctly and it has been a while.
But after having a closer look, it seems that this method can only be called after `resolve` once run through `ifRequestedButUnresolved` with `createTestDescriptor`.
If this is true, the whole error-handling in this method can be removed, can't it?